src/components/ModularTable/ModularTable.tsx

```diff
@@ -65,7 +65,58 @@ export type Props<D extends Record<string, unknown>> = PropsWithSpread<
   HTMLProps<HTMLTableElement>
 >;
 
-function ModularTable({
+const generateCell = <D extends Record<string, unknown>>(
+  cell: Cell<D>,
+  getCellProps: Props<D>["getCellProps"]
+) => {
+  const hasColumnIcon = cell.column.getCellIcon;
+  const iconName = hasColumnIcon && cell.column.getCellIcon?.(cell);
+
+  return (
+    <TableCell
+      {...cell.getCellProps([
+        {
+          className: cell.column.className,
+        },
+        {
+          className: hasColumnIcon ? "p-table__cell--icon-placeholder" : "",
+        },
+        { ...getCellProps?.(cell) },
+      ])}
+    >
+      {iconName && <Icon name={iconName} />}
+      {cell.render("Cell")}
+    </TableCell>
+  );
+};
+
+const generateRows = <D extends Record<string, unknown>>(
+  rows: Row<D>[],
+  prepareRow: (row: Row<D>) => void,
+  getRowProps: Props<D>["getRowProps"],
+  getCellProps: Props<D>["getCellProps"]
+) => {
+  let tableRows: ReactNode[] = [];
+  rows.forEach((row) => {
+    // This function is responsible for lazily preparing a row for rendering.
+    // Any row that you intend to render in your table needs to be passed to this function before every render.
+    // see: https://react-table.tanstack.com/docs/api/useTable#instance-properties
+    prepareRow(row);
+    tableRows.push(
+      <TableRow {...row.getRowProps(getRowProps?.(row))}>
+        {row.cells.map((cell) => generateCell<D>(cell, getCellProps))}
+      </TableRow>
+    );
+    if (row.subRows?.length) {
+      tableRows = tableRows.concat(
+        generateRows<D>(row.subRows, prepareRow, getRowProps, getCellProps)
+      );
+    }
```

Comment:
This is the only change to support subRows, the rest of the changes were moving the row generation to a function so that the main rows and sub-rows can both be generated by the same func.